lib/core/src/Cardano/Wallet/Network.hs

```diff
@@ -106,24 +106,18 @@ data NetworkLayer m target block = NetworkLayer
         -- If the node has adopted an alternate fork of the chain, it will
         -- return 'RollBackward' with a new cursor.
 
-    , findIntersection
-        :: Cursor target -> m (Maybe BlockHeader)
-        -- ^ Attempt to find an intersection between the node's unstable blocks
-        -- and a given list of headers. This can be useful if we need to know
-        -- whether we are 'in sync' with the node or, close enough.
-
     , initCursor
-        :: [BlockHeader] -> Cursor target
+        :: [BlockHeader] -> m (Cursor target)
         -- ^ Creates a cursor from the given block header so that 'nextBlocks'
         -- can be used to fetch blocks.
 
     , cursorSlotId
         :: Cursor target -> SlotId
         -- ^ Get the slot corresponding to a cursor.
 
-    , networkTip
-        :: ExceptT ErrNetworkTip m BlockHeader
-        -- ^ Get the current network tip from the chain producer
+    , currentNodeTip
```

Comment:
Nice, you renamed it 👍 

Wanted to say though:
- We seem to use `currentNodeTip` only for:
    - `GetNetworkInformation`
    -  `monitorStakePools` (in a jörmungandr influenced way)
- In this PR `currentNodeTip` for Byron seem to return the  **current point** of the in-progress chain-sync, right?

> have one network client per wallet (there's one global network layer, but that can't work with the stateful haskell nodes).

What network-layer will you use for the `NetworkInformation`-endpoint?

I wonder if `currentNodeTip` can not be removed or implemented in another way than the chain-sync protocol.

Reply:
`currentNodeTip` in the byron layer returns the tip of the underlying node. Which is what we want. The cursor tells you where you're at.

Reply:
Hm… Ah! You're right.